### PR TITLE
add group-by-modifier chapter

### PR DESCRIPTION
**Pull request by AilinKid**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/group-by-modifiers.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
+---
+title: GROUP BY Modifiers
+summary: Learn about the group by modifiers in TiDB.
+aliases: ['/docs/dev/functions-and-operators/group-by-modifiers/','/docs/dev/reference/sql/functions-and-operators/group-by-modifiers/']
```
Comment (qiancai) [inline functions-and-operators/group-by-modifiers.md:4]:
```suggestion
```

Comment (qiancai) [reply functions-and-operators/group-by-modifiers.md:4]: aliases 用于替换某篇要删除的文档

File: functions-and-operators/group-by-modifiers.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
+---
+title: GROUP BY Modifiers
```
Comment (qiancai) [inline functions-and-operators/group-by-modifiers.md:2]: 对于新增文档，需要加到文档目录里，请修改下 TOC.md 以及 TOC-cloud.md

File: functions-and-operators/group-by-modifiers.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+summary: Learn about the group by modifiers in TiDB.
+aliases: ['/docs/dev/functions-and-operators/group-by-modifiers/','/docs/dev/reference/sql/functions-and-operators/group-by-modifiers/']
+---
+# GROUP BY Modifiers
+
+## Normal Usage Guide
+
```
Comment (qiancai) [inline functions-and-operators/group-by-modifiers.md:9]:
```suggestion
```

File: functions-and-operators/group-by-modifiers.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
++------+---------+---------+--------+
+3 rows in set (0.002 sec)
+```
+
+With ROLLUP added, the query produces several extra rows:
+
+```sql
+TiDB [test]> SELECT year, country, product, SUM(profit) AS profit
+    ->        FROM sales
+    ->        GROUP BY year, country, product WITH ROLLUP;
```
Comment (windtalker) [inline functions-and-operators/group-by-modifiers.md:86]: Maybe add a order by clause like `order by year desc,country desc, product desc` wil make it easier to understand the following explainations?

File: functions-and-operators/group-by-modifiers.md [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+For `GROUP BY ... WITH ROLLUP` queries, to test whether `NULL` values in the result represent super-aggregate values, the `GROUPING()` function is available for use in the select list, `HAVING` clause, and ORDER BY clause. For example, `GROUPING(year)` returns 1 when `NULL` in the year column occurs in a super-aggregate row, and 0 otherwise. Similarly, `GROUPING(country)` and `GROUPING(product)` return 1 for super-aggregate `NULL` values in the country and product columns, respectively:
+
+```sql
+TiDB [test]> SELECT
+    ->          year, country, product, SUM(profit) AS profit,
+    ->          GROUPING(year) AS grp_year,
+    ->          GROUPING(country) AS grp_country,
+    ->          GROUPING(product) AS grp_product
+    ->        FROM sales
+    ->        GROUP BY year, country, product WITH ROLLUP;
```
Comment (windtalker) [inline functions-and-operators/group-by-modifiers.md:126]: Ditto

File: functions-and-operators/group-by-modifiers.md [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+| hoop      | NULL      |        3 |
+| ball      | All sizes |       35 |
+| hoop      | small     |       15 |
+| hoop      | large     |        5 |
+| hoop      | All sizes |       23 |
++-----------+-----------+----------+
+9 rows in set (0.029 sec)
+```
+
+## How to read the plan detail from ROLLUP?
```
Comment (windtalker) [inline functions-and-operators/group-by-modifiers.md:259]: Does this part need to added in user docs?

**Comment by AilinKid** [comment]
migrate to https://github.com/pingcap/docs-cn/pull/15053

**Pull request closed without merging**